ourjapanlife/findadoc-server
# Build out validation for healthcare professional names

Resolves #337 

# What changed
Previously the validation for `createHealthcareProfessional ` names field where incomplete. In this PR I have added more validations.

1. Created a new folder for validations.
2. Created a validationHealthcareProfessional file.
3. Added more validations to `validateNames` function.
4. Added more` ErrorCode's` in result file.
5. Refactor the tests for `createHealthcareProfessional`.

# Testing instructions

1. Should create an error when there is a duplicate local:

```
"names": [
      {
        "firstName": "John",
        "lastName": "Doe",
        "locale": "en_US"
      },
      {
        "firstName": "Jane",
        "lastName": "Doe",
        "locale": "en_US"
      }
    ]
```


2. Should create an error when local en has Japanese characters and vice versa:

```
"names": [
      {
        "firstName": "ジョン",
        "lastName": "Doe",
        "locale": "en_US"
      }
    ]
```
Or
```
"names": [
      {
        "firstName": "ジョン",
        "lastName": "Doe",
        "locale": "ja_JP"
      }
    ]
```

3. Should create an error when name contains invalid characters:

```
"names": [
      {
        "firstName": "John",
        "lastName": "Doe",
        "middleName": "Edward9",
        "locale": "en_US"
      }
    ]
```

4. Should create an error when names are to long:

```
"names": [
      {
        "firstName": "John9",
        "lastName": "Thisisaverylongnamethatwillcauseanerror",
        "locale": "en_US"
      }
    ]
```

# To do

- Write tests for the validations
